### ISSUE TITLE
Implement support for scrolling through the whole remote dataset

### ISSUE THREAD

**Issue by roblotter** [infinite-table/infinite-react#28]
We need to add support for scrolling through entire remote datasets - the server will tell `InfiniteTable` the size of the dataset, and `InfiniteTable` needs to render all those rows, even they are not entirely loaded yet.

**Comment by roblotter** [comment]
## WIP Initial proposal

### When there is grouping

```tsx
const data = {
   groupRows: [
      {
        groupKey: 'Management',
        groupRowsCount: 80, // have 80 child items in the DB for this group key 
       // needed in order to know how many group rows to render
       // although just the first 10 are present now - see the `groupRows` array

        groupDataCount: 2400, // this is the count of all actual data items (rows, so only leafs) 
        groupRows: [
           {
             groupKey: 'USA',
             groupData: [....], // have 100 data items initially, although in this group, there are 1400 items in database
             groupDataCount: 1400
           },
           {
             groupKey: 'UK',
             groupData: [....], // have 80 data items initially, although in this group, there are 2000 items in database
             groupDataCount: 2000
           },
           //.... another 8 child groups here
        ]
      }
   ]
}
<DataSource data={data} />
```

**Comment by roblotter** [comment]
The structure proposed above is deeply nested - another option would be a more de-normalized one, like below (still early exploration)

```tsx
const data = {
  groups: {
    keys: [
      ['Management'],
      ['Management','TopLevel'],
      ['IT'],
      // initially just those 3 groups should be rendered
    ],
    keysCount: 200, // 200 total group rows on server
    groupValues: {
      'Management/TopLevel': {
        data: [ 
            // initially say we have 100 records, while the groupCount is 500 - items on server
        ],
        groupCount: 500
      },
    }
  } ,
}
```

**Comment by roblotter** [comment]
This was implemented in v `0.2.3`, so closing this
